android/java/org/chromium/chrome/browser/bookmarks/BraveBookmarkManagerMediator.java

```diff
@@ -130,21 +130,37 @@ private void doImportBookmarks() {
         Intent intent = new Intent(Intent.ACTION_GET_CONTENT);
         intent.setType("text/html");
         intent.addCategory(Intent.CATEGORY_OPENABLE);
-        if (mWindowAndroid.showIntent(Intent.createChooser(intent,
-                                              mContext.getResources().getString(
-                                                      R.string.import_bookmarks_select_file)),
-                    new WindowAndroid.IntentCallback() {
-                        @Override
-                        public void onIntentCompleted(int resultCode, Intent results) {
-                            if (resultCode == Activity.RESULT_OK && results != null
-                                    && results.getData() != null) {
-                                PostTask.postTask(TaskTraits.USER_VISIBLE_MAY_BLOCK,
-                                        () -> { importFileSelected(results.getData()); });
-                            }
+        if (mWindowAndroid.showIntent(
+                Intent.createChooser(
+                        intent,
+                        mContext.getResources().getString(R.string.import_bookmarks_select_file)),
+                new WindowAndroid.IntentCallback() {
+                    @Override
+                    public void onIntentCompleted(int resultCode, Intent results) {
+                        if (resultCode == Activity.RESULT_OK
+                                && results != null
+                                && results.getData() != null) {
+                            PostTask.postTask(
+                                    TaskTraits.USER_VISIBLE_MAY_BLOCK,
+                                    () -> {
+                                        importFileSelected(results.getData());
+                                    });
                         }
-                    },
-                    null))
+                    }
+                },
+                null)) {
             return;
+        }
+    }
+
+    private String sanitizeFilename(String displayName) {
+        String[] badCharacters = new String[] {"..", "/"};
+        String[] segments = displayName.split("/");
+        String fileName = segments[segments.length - 1];
```

Comment:
1) can we check that segments is not empty?
2) why do we replace "/" after splitting the string with "/" - probably can just check that  fileName has no slashes
3) overall it's not clear how smart this function should be.. i.e. why don't we drop single dots or control characters

Reply:
cc: @AlexeyBarabash

Reply:
@iefremov 

I was driven by `[UnsanitizedFilenameFromContentProvider]` lint error and  I wasn't able to find the the requirements for this function.
The general idea that we cannot trust the filename which came from a ContentProvider and use it without proper sanitization. 
Unsanitized filenames may contain characters or paths that pose security risks, such as directory traversal.

> overall it's not clear how smart this function should be.. i.e. why don't we drop single dots or control characters

totally true

The idea for the `sanitizedName` is to use it to save a temporary file at our data directory.
So we are free to chose any name we want, and then we don't need to sanitize anything.

Fixed at https://github.com/brave/brave-core/pull/25340/commits/2515f5e920d9e9ca4a68e0c4c0b6ac33ba243edd.

Thanks for bringing my attention here.